Improve api doc tests by eliminating 218 Zephyr-related checks

showell opened the pull request:
I also do some general cleanup.

laurynmm reviewed (commented):
@showell - I think these updates look good, though other folks would know more about the zephyr-related checks than I do.

I had one small comment about where we might be able to document needing to update the test if one adds a new API endpoint to the documentation.

showell commented on zerver/tests/test_docs.py:
```diff
@@ -128,9 +128,10 @@ def test_api_doc_endpoints(self) -> None:
         # Validate that the parsing logic isn't broken, since if it
         # broke, the below would become a noop.
-        self.assertGreater(len(endpoint_list), 70)
+        self.assert_length(endpoint_list, 110)
```
It's useful when looking at a test to know exactly how expensive it is.

showell replied:
I'll just make the number more realistic (but imprecise), so that future people like me that profile the test get a quicker idea of how many endpoints there are.

showell replied:
If you simply assert that the list is non-zero, you still have the danger that some change to the regex or url structure or way of putting the links into the main document silently removes coverage.

timabbott replied:
Well, you can change the number to asserting it's at least 110. I just can guarantee it'll be annoying to update every time we add a new documented API endpoint, and I think that annoyingness outweighs any downside for this number being slightly imprecise.

showell commented:
This is ready for review again.  I addressed the overly precise check for 110 endpoints and moved a better version of that commit to the end of the PR.

I also added a commit to remove some redundant checks.

timabbott commented:
Nice, marked to merge once CI passes, thanks @showell!
